mlflow/pytorch.py

```diff
@@ -209,14 +240,22 @@ def load_model(path, run_id=None, **kwargs):
     if run_id is not None:
         path = mlflow.tracking.utils._get_model_log_dir(model_name=path, run_id=run_id)
     path = os.path.abspath(path)
-    flavor_conf = _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME)
 
-    if torch.__version__ != flavor_conf["pytorch_version"]:
+    try:
+        pyfunc_conf = _get_flavor_configuration(model_path=path, flavor_name=pyfunc.FLAVOR_NAME)
```

Comment:
This adds a dependency on pyfunc flavor. I guess it makes sense in this case to add the dependency rather than store the code path data twice. but since we already have the dependency would it make sense to just call _load_pyfunc here and return the model from the wrapper? Should we move the version check from load_model to _load_model?

Reply:
After discussing offline, version validation cannot currently be performed via the `_load_pyfunc` code path because the path to the PyTorch model directory (the **data path**) is passed to `_load_pyfunc`, instead of the model root directory. This prevents us from accessing and parsing the `MLmodel` configuration and should be addressed in the future.